btcec/pubkey.go

```diff
@@ -54,6 +54,24 @@ const (
 	pubkeyHybrid       byte = 0x6 // y_bit + x coord + y coord
 )
 
+// IsCompressedPubKey returns true the the passed serialized public key has
+// been encoded in compressed format, and false otherwise.
+func IsCompressedPubKey(pubKey []byte) bool {
+	// If the length of the public key isn't 33-bytes, then we can rule it
+	// out early.
+	if len(pubKey) != PubKeyBytesLenCompressed {
+		return false
+	}
+
+	// Next, extract the format that's encoded within the first byte of the
+	// serialized pubkey.
+	format := pubKey[0]
+
+	// With the format isoluated it MUST be of a compressed key type,
```

Comment:
isolated

That said, while there is nothing wrong with this function as is, it could be a bit more compact and avoid the magic numbers as follows:

```Go
	// The public key is only compressed if it is the correct length and the
	// format (first byte) is one of the compressed pubkey values.
	return len(pubKey) == PubKeyBytesLenCompressed &&
		(pubKey[0]&^byte(0x1) == pubkeyCompressed)
```